### ISSUE TITLE
[Feature request] Make support for KDE apps in Fleet Commander

### ISSUE THREAD

**Issue by MurzNN** [fleet-commander/fc-admin#155]
Will be good to have support for KDE applications configs in Fleet Commander, located in ~/.kde/config folder too, additionally to Gnome.

**Comment by olivergs** [comment]
The goal of fleet commander is not to mess with User's home directory files.

For GNOME applications, fleet commander makes use of gsettings / dconf and creates a layer over base configuration and below user configuration in that specific machine, so the settings specified by the administrator overrides the system defaults, and the user specific settings for that machine overrides even the admin ones.

As long as KDE (AFAIK) only writes configuration in text files in home directory, it is not something Fleet Commander will handle right now.

**Comment by MurzNN** [comment]
KDE have same configuration tool like GNOME, named `kwriteconfig` - https://userbase.kde.org/KDE_System_Administration/Configuration_Files

**Comment by olivergs** [comment]
Of course that tool exist. But what it does is to write directly on home directory config files for that user. Also it does not care of what was written already there, and it does not give the ability for that user to override a setting.

Every time the user logs in that machine will have to change the setting again because fleet commander has no mechanism to keep changes over sessions, and that feature is not intended to be added to fleet commander.

**Comment by aruiz** [comment]
For Fleet Commander to work with KDE, KDE would have to adopt dconf, or a similar system that has configuration change notification and a layered backend that allows us to put the profile db in /run. 

Otherwise we would have to do a very intrusive operation (walking into HOME and writing files over there) which can result in all sorts of conflicts and possible corruption problems.

**Comment by aleixpol** [comment]
I've just randomly passed by this ticket through another one.

For what it's worth, it's 100% possible to make KConfig look at another directory that goes between /etc and ~/.config. This fallback mechanism has been in place for decades and it is something we rely on for different use-cases.

**Comment by aruiz** [comment]
Hey @aleixpol,

Thanks for chiming ing, it is indeed useful to confirm that layering is an option.

Are there notifications forr changes in specific configuration keys though? We need to subscribe to individual changes for the system to work.

**Comment by aleixpol** [comment]
Yes, here's an example of such notifications through `dbus-monitor`

```
signal time=1680561890.915196 sender=:1.2952 -> destination=(null destination) serial=173 path=/kwinrc; interface=org.kde.kconfig.notify; member=ConfigChanged
   array [
      dict entry(
         string "Wayland"
         array [
            array of bytes "EnablePrimarySelection"
         ]
      )
   ]
```

I guess some more documentation on `org.kde.kconfig.notify` if that was useful, tell us and we'll see to getting this polished.

**Comment by aruiz** [comment]
Nice!

I will reopen this bug then, this is certainly feasable. On the maintainers side we're unlikely to implement this ourselves but if someone steps up we'll gladly give guidance and reviews.
